src/include/maths_utils.h

```diff
@@ -37,5 +37,6 @@
 #include <stdint.h>
 
 uint8_t ulog2(uint32_t value);
+uint32_t swd_odd_parity(const uint32_t value);
```

Comment:
The `const` here can be dropped (clang-tidy should be linting about this) from here (but not the definition in maths_utils.c).

Perhaps lets use `calc_odd_parity` or `calculate_odd_parity` as a more generic term for what this is doing as it's non-specific to SWD and that'll keep things open in the nomenclature for the JTAG-PDI support which will definitely benefit from this?

Reply:
`calculate_odd_parity` is good, but too verbose -- all math functions "calculate". To help IntelliSense and likes, I suggest `parity_odd` to put the context first. I wonder if `parity_even` exists elsewhere.
The `const`, to my understanding, should be visible from headers so that the compiler can optimize other translation units to not duplicate arguments into r0 (or whatever the AAPCS first argument register is). Or are we only using it for guarantees on pointers?

Reply:
`parity_odd` reads.. strangely.. to us, due to us referencing here calculating the parity value for an odd parity scheme - it's technically correct, but it's got the same odd lilt as Picard saying "Tea, Earl Grey, Hot, Black" rather than "a hot cup of black Earl Grey tea".

The `const` on a value in a C function declaration (prototype) does nothing, and is purely noise - this is what the clang-tidy lint is about. It only has meaning in the function definition.

In regards pointers, `const char *` is saying the thing the pointer points to is const, but `const char *const` doesn't do anything extra in the context of a decl. That's why we omit on the parameter side of the * in the decl.

Reply:
Renamed `swd_odd_parity` to `calculate_odd_parity` globally.
Removed `const` from uint32_t in this header.